src/evalvars.c

```diff
@@ -1158,7 +1158,9 @@ ex_let(exarg_T *eap)
     // declaration, not the expression.
     SOURCING_LNUM = cur_lnum;
 
-    if (!eap->skip && eval_res != FAIL)
+    if (!eap->skip && eval_res != FAIL && rettv.v_type == VAR_CLASS)
```

Comment:
You can use list assignment to assign multiple variables (B is a class in the statement below):

```
    [a, b] = [10, B]
```

To handle this case, this check should be moved to the ex_let_one() function.

Reply:
> To handle this case, this check should be moved to the ex_let_one() function

Thanks, it was still on my list.